Tests/Marketplace/search_and_install_packs.py

```diff
@@ -372,13 +372,23 @@ def install_packs(client: demisto_client,
 
                     # We've more attempts, retrying without tho malformed packs.
                     SUCCESS_FLAG = False
-                    logging.error(f"Unable to install malformed packs: {malformed_ids}, retrying without them.")
                     packs_to_install = [pack for pack in packs_to_install if pack['id'] not in malformed_ids]
+                    if packs_to_install:
+                        logging.error(f"Unable to install malformed packs: {malformed_ids}, retrying without them.")
+                    else:
+                        raise Exception(f"Unable to install malformed packs: {malformed_ids}, no other packages left to install")
 
                 if (error_ids := get_error_ids(ex.body)) and WLM_TASK_FAILED_ERROR_CODE in error_ids:
-                    # If we got this error code, it means that the modeling rules are not valid, exiting install flow.
-                    raise Exception(f"Got [{WLM_TASK_FAILED_ERROR_CODE}] error code - Modeling rules and Dataset validations "
-                                    f"failed. Please look at GCP logs to understand why it failed.") from ex
+                    if len(packs_to_install) == 1:
+                        # If we only have one pack, and it failed, there is noting else to try, exiting.
+                        raise Exception(f"Got [{WLM_TASK_FAILED_ERROR_CODE}] error code - Modeling rules and Dataset validations "
+                                        f"failed. Please look at GCP logs to understand why it failed.") from ex
+
+                    # We've multiple packs to install, retrying one pack at a time.
+                    logging.debug(f"Failed to install packs:{packs_to_install}, trying one pack at a time, with only one attempt")
+                    for pack_to_install in packs_to_install:
+                        SUCCESS_FLAG &= install_packs(client, host, [pack_to_install], request_timeout, sleep_interval,
+                                                      attempts_count=1)
```

Comment:
Why retry only if WLM_TASK_FAILED_ERROR_CODE happens and not change it in the prepared batch with dependencies? 

Here it looks like plaster for only specific issues, but changing it when creating a batch with dependencies may prevent future bugs.

Reply:
it's no different from handling "malformed_ids", only here we don't know which one failed, so we try them one-by-one.

To add it to the batches will cause multiple installs of the sub-dependencies which is not efficient. 

The optimal solution is to get the exact error from the server on which pack failed.